lib/devise_two_factorable/models/two_factorable.rb

```diff
@@ -127,6 +137,10 @@ def generate_otp_persistence_seed
       self.otp_persistence_seed = SecureRandom.hex
     end
 
+    def valid_drift_window?
+      (self.last_successful_otp_at - Time.now).abs < self.class.otp_drift_window
```

Comment:
Aha, I _think_ I see now...
I think the name `valid_drift_window?` might be a misnomer. Isn't this really `token_already_used?` (or similar/inversed)
Additionally, this method takes in the responsibility of `self.last_successful_otp_at &&` because if there's no successful, this function should represent that